### PR TITLE
changes for reset otp api

### PR DESCRIPTION
**Pull request by saravanpa-aot**
*Issue #:*
https://github.com/bcgov/entity/issues/3239

*Description of changes:*

*Checklist:*
I confirm that below checklist items are addressed in this pull request; (check the boxes applicable)
- [ ] No lint errors
- [ ] No test case failures and proper coverage for all modules/classes
- [ ] Updated the deployment configs for new environment variable(s)
- [ ] Updtaed the postman collection in entity repository (https://github.com/bcgov/entity/tree/master/api-e2e/postman) for e2e tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

**Review by nitheesh-aot — APPROVED** [review]

File: auth-api/src/auth_api/resources/user.py [hunk 1]
```diff
@@ -138,6 +138,32 @@ def get():
         return response, status
 
 
+@cors_preflight('GET,OPTIONS,DELETE')
+@API.route('/<path:username>/otp', methods=['GET', 'OPTIONS', 'DELETE'])
```
Comment (sumesh-aot) [inline auth-api/src/auth_api/resources/user.py:142]: Is it user name? username have different mapping in different environment (not by design). Some env may have `/` in username. Did you try this whether it works for all cases ?

Comment (saravanpa-aot) [reply auth-api/src/auth_api/resources/user.py:142]: we are using username in other apis as well..most of the user maangement apis by staff/others has this pattern..
the specific / is addressed by the path i guess... i think we discussed this and found the path works..

Comment (pwei1018) [reply auth-api/src/auth_api/resources/user.py:142]: Is the username be unique key in db?

Comment (saravanpa-aot) [reply auth-api/src/auth_api/resources/user.py:142]: Its not the primary key..but i guess it will be unique..

**Review by sumesh-aot — APPROVED** [review]

**Comment by sumesh-aot** [comment]
@saravanpa-aot  `multiple_heads = "Multiple head revisions are present for given argument 'head'; please specify a specific target revision, '<branchname>@head' to narrow to a specific head, or 'heads' for all heads"
`

**Comment by saravanpa-aot** [comment]
> @saravanpa-aot `multiple_heads = "Multiple head revisions are present for given argument 'head'; please specify a specific target revision, '<branchname>@head' to narrow to a specific head, or 'heads' for all heads" `

oh..didnt get this in local... let me check

**Review by pwei1018 — APPROVED** [review]

**Pull request merged**